apps/designer/app/dashboard/projects/create-rename-duplicate-delete.tsx

```diff
@@ -0,0 +1,276 @@
+        errors={errors}
+        placeholder="New Project"
+        label="Project Title"
+        primaryButton={<Button type="submit">Create Project</Button>}
+      />
+    </Dialog>
+  );
+};
+
+const useRenameProject = ({
+  projectId,
+  onComplete,
+}: {
+  projectId: DashboardProject["id"];
+  onComplete: () => void;
+}) => {
+  const { submit, data } = trpc.rename.useMutation();
+
+  useEffect(() => {
+    if (data === undefined || "errors" in data) {
+      return;
+    }
+    onComplete();
+  }, [data, onComplete]);
+
+  const handleSubmit = ({ title }: { title: string }) => {
+    submit({ projectId, title });
+  };
+
+  return {
+    handleSubmit,
+    errors: data && "errors" in data ? data.errors : undefined,
+  };
+};
+
+export const RenameProject = ({
+  isOpen,
+  title,
+  projectId,
+  onComplete,
+  onOpenChange,
+}: {
+  isOpen: boolean;
+  title: string;
+  projectId: DashboardProject["id"];
+  onComplete: () => void;
+  onOpenChange: (isOpen: boolean) => void;
+}) => {
+  const { handleSubmit, errors } = useRenameProject({ projectId, onComplete });
+  return (
+    <Dialog title="Rename" isOpen={isOpen} onOpenChange={onOpenChange}>
+      <Content
+        onSubmit={handleSubmit}
+        errors={errors}
+        title={title}
+        label="Project Title"
+        primaryButton={<Button type="submit">Rename Project</Button>}
+      />
+    </Dialog>
+  );
+};
+
+const useDeleteProject = ({
+  projectId,
+  title,
+  onComplete,
+}: {
+  projectId: DashboardProject["id"];
+  title: string;
+  onComplete: () => void;
+}) => {
+  const { submit, data } = trpc.delete.useMutation();
+  const [isMatch, setIsMatch] = useState(false);
+
+  useEffect(() => {
+    if (data === undefined || "errors" in data) {
+      return;
+    }
+    onComplete();
+  }, [data, onComplete]);
+
+  const handleSubmit = () => {
+    submit({ projectId });
+  };
+
+  const handleChange = ({ title: currentTitle }: { title: string }) => {
+    setIsMatch(currentTitle.trim().toLowerCase() === title.toLowerCase());
+  };
+
+  return {
+    handleSubmit,
+    handleChange,
+    errors: data && "errors" in data ? data.errors : undefined,
+    isMatch,
+  };
+};
+
+export const DeleteProject = ({
+  isOpen,
+  title,
+  projectId,
+  onComplete,
+  onOpenChange,
+}: {
+  isOpen: boolean;
+  title: string;
+  projectId: DashboardProject["id"];
+  onComplete: () => void;
+  onOpenChange: (isOpen: boolean) => void;
+}) => {
+  const { handleSubmit, handleChange, errors, isMatch } = useDeleteProject({
+    projectId,
+    onComplete,
+    title,
+  });
+  return (
+    <Dialog
+      title="Delete Confirmation"
+      isOpen={isOpen}
+      onOpenChange={onOpenChange}
+    >
+      <Content
+        onSubmit={handleSubmit}
+        onChange={handleChange}
+        errors={errors}
+        label={
+          <>
+            Confirm by typing{" "}
+            <Text as="span" color="error" variant="label">
+              {title}
+            </Text>{" "}
```

Comment:
It's evil that I can't select and copy the title (Text has `userSelect: "none"`) 😄 

If this is intentional to make 100% sure user isn't deleting a wrong project, I still think this is wrong: if I'm copy-pasting it's on me

Reply:
BTW, why `userSelect: "none"` is default for Text? 

I often want to select text everywhere. For example, if I'm telling someone via a chat what field to fill, I often want to select a copy the label directly from UI. Maybe I'm weird like that, but it always bothers me when developers make text unselectable for no reason.

Reply:
not intentional

Reply:
I think it was meant to solve the problem in places where you accidentally select text, but I forgot what the exact use case was, removed it now